include/engine/guidance/overview_assembler.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef ENGINE_GUIDANCE_OVERVIEW_ASSEMBLER_HPP
+#define ENGINE_GUIDANCE_OVERVIEW_ASSEMBLER_HPP
+
+#include "engine/guidance/leg_geometry.hpp"
+#include "engine/douglas_peucker.hpp"
+
+#include <boost/range/irange.hpp>
+
+#include <vector>
+
+namespace osrm
+{
+namespace engine
+{
+namespace guidance
+{
+class OverviewAssembler
+{
+  public:
+    std::vector<util::FixedPointCoordinate>
+    operator()(const std::vector<LegGeometry> &leg_geometries, const unsigned zoom_level) const
```

Comment:
Yep, transforming these. Doesn't make much sense when you don't capture the Datafacade.